source/common/buffer/zero_copy_input_stream_impl.h

```diff
@@ -14,28 +14,33 @@ namespace Buffer {
 class ZeroCopyInputStreamImpl : public virtual google::protobuf::io::ZeroCopyInputStream {
 public:
   // Create input stream with one buffer, and finish immediately
-  ZeroCopyInputStreamImpl(Buffer::Instance& buffer);
+  ZeroCopyInputStreamImpl(Buffer::InstancePtr&& buffer);
 
   // Create input stream with empty buffer
-  ZeroCopyInputStreamImpl() {}
-
-  virtual ~ZeroCopyInputStreamImpl() {}
+  ZeroCopyInputStreamImpl();
 
   // Add a buffer to input stream, will consume all buffer from parameter
   // if the stream is not finished
   void move(Buffer::Instance& instance);
 
-  // Mark the buffer is finished
+  // Mark the stream is finished
   void finish() { finished_ = true; }
 
   // google::protobuf::io::ZeroCopyInputStream
+  // See
+  // https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.io.zero_copy_stream#ZeroCopyInputStream
+  // for each methods details.
+
+  // Note the Next() will return true with no data until next data available if the stream is not
```

Comment:
// Note Next() will return true with no data until more data is available if the stream is not
// finished. It is the caller's responsibility to finish the stream or wrap with LimitingInputStream
// before passing to protobuf code to avoid a spin loop.